src/helpers/messaging.rs

```diff
@@ -153,7 +159,7 @@ pub struct GatewayConfig {
 impl Gateway {
     pub fn new<N: Network>(role: Role, network: &N, config: GatewayConfig) -> Self {
         let (tx, mut receive_rx) = mpsc::channel::<ReceiveRequest>(1);
-        let (envelope_tx, mut envelope_rx) = mpsc::channel::<(ChannelId, MessageEnvelope)>(1);
+        let (envelope_tx, mut envelope_rx) = mpsc::channel::<SendRequest>(1);
```

Comment:
Should we be configuring the size of the buffers on channels?  For tests, 1 is fine, but it seems like maybe we should let them hold a few more otherwise.

Not for this change, but it seems like we might benefit from it some time.

Reply:
let me create a task for it